crates/oxc_parser/src/js/expression.rs

```diff
@@ -398,7 +398,13 @@ impl<'a> ParserImpl<'a> {
         let value = self.cur_string();
         let span = self.start_span();
         self.bump_any();
-        Ok(self.ast.string_literal(self.end_span(span), value))
+        let span = self.end_span(span);
+        // SAFETY:
+        // range comes from the parser, which are ensured to meeting the criteria of `get_unchecked`.
+        let raw = Atom::from(unsafe {
+            self.source_text.get_unchecked(span.start as usize..span.end as usize)
+        });
```

Comment:
Are we sure we can rely on correctness of whatever is generating the `Span` enough to use `get_unchecked`?

Also, comment says "range comes from the parser". This code is in the parser. Should it be "range comes from the *lexer*"?